glom/test/test_grouping.py

```diff
@@ -0,0 +1,44 @@
+from glom.grouping import Group, First, Avg, Sum, Max, Min, Count, Sample, Limit
+from glom import glom, T
+
+from glom.reduction import Merge, Flatten
+
+
+def test_bucketing():
+    assert glom(range(4), Group({lambda t: t % 2 == 0: [T]})) == {True: [0, 2], False: [1, 3]}
+    assert (glom(range(6), Group({lambda t: t % 3: {lambda t: t % 2: [lambda t: t / 10.0]}})) ==
+        {0: {0: [0.0], 1: [0.3]}, 1: {1: [0.1], 0: [0.4]}, 2: {0: [0.2], 1: [0.5]}})
+
+
+def test_agg():
+    t = list(range(10))
+    assert glom(t, Group(First())) == 0
+    assert glom(t, Group(T)) == 9  # this is basically Last
+
+    assert glom(t, Group(Avg())) == 1.0 * sum(t) / len(t)
```

Comment:
for future reference, you can import the division future to avoid the multiplication by 1.0